pkg/controllers/externalsecret/externalsecret_controller_test.go

```diff
@@ -288,7 +313,7 @@ var _ = Describe("ExternalSecret controller", func() {
 			Expect(secret.ObjectMeta.Labels).To(BeEquivalentTo(es.Spec.Target.Template.Metadata.Labels))
 			Expect(secret.ObjectMeta.Annotations).To(BeEquivalentTo(es.Spec.Target.Template.Metadata.Annotations))
 
-			cleanEs := tc.externalSecret.DeepCopyObject()
+			cleanEs := (tc.externalSecret.DeepCopyObject()).(*esv1alpha1.ExternalSecret)
```

Comment:
the interface of `client.MergeFrom` below was changed in https://github.com/kubernetes-sigs/controller-runtime/commit/a79f6f1353572a1916393f0b68605719a10451f8. We can not use a `runtime.Object` anymore because it doesn't implement the `metav1.Object` interface (`GetAnnotations()` etc). Using a type assertion seems like the easiest way.

Reply:
nvm; `DeepCopy()` returns the ES